master/pkg/searcher/custom_search.go

```diff
@@ -3,17 +3,22 @@ package searcher
 import (
 	"encoding/json"
 
+	"google.golang.org/protobuf/encoding/protojson"
+
 	"github.com/determined-ai/determined/master/pkg/model"
 	"github.com/determined-ai/determined/master/pkg/schemas/expconf"
 	"github.com/determined-ai/determined/proto/pkg/experimentv1"
+
+	"github.com/pkg/errors"
 )
 
 type (
 	customSearchState struct {
-		// store the operations
-		// store the events
-		SearchMethodType     SearchMethodType `json:"search_method_type"`
-		SearcherEventQueue   *SearcherEventQueue
+		SearchMethodType   SearchMethodType `json:"search_method_type"`
+		SearcherEventQueue *SearcherEventQueue
+		// a list of marshaled protobuf experimentv1.Events object.
+		// Each protobuf obj needs to be marshaled individually.
+		PBEventsJSON         []json.RawMessage
 		customSearchProgress float64
```

Comment:
I think the progress is also going to need to be exported and serialized, so that we have a value after restoration and before the next trial progress event comes in.

Though, after seeing this, I think I would propose doing something different altogether, where we have a separate type used only for serialization, and do a quick translation between the two. I think that would be nicer than of muddying up the main type with the field `PBEventsJSON` field.